src/NServiceBus.Core/Pipeline/Incoming/ReceiveFeature.cs

```diff
@@ -65,6 +66,9 @@ public Task<CompletableSynchronizedStorageSession> TryAdapt(TransportTransaction
             internal static readonly Task<CompletableSynchronizedStorageSession> EmptyResult = Task.FromResult<CompletableSynchronizedStorageSession>(new NoOpCompletableSynchronizedStorageSession());
         }
 
+        // Do not ally Fody to weave the IDisposable for us so that other threads can still access the instance of this class
```

Comment:
@WilliamBZA u r correct. i dug into the code a bit more and it seems it is only disposable for downstreams so a no op is fine